test/unittests/UtilityTest.php

```diff
@@ -873,4 +1049,261 @@ public function testPathJoin(): void
+     *
+     * @covers Utility::toDateDisplayFormat
+     * @return void
+     */
+    public function testToDateDisplayFormatThrowsException()
+    {
+        $this->_setMockDB();
+        $date = "2000-01-";
+        $this->expectException('\LorisException');
+        Utility::toDateDisplayFormat($date);
+    }
+
+    /**
+     * Test that getScanTypeList returns the scan type information
+     * in the correct format
+     *
+     * @covers Utility::getScanTypeList
+     * @return void
+     */
+    public function testGetScanTypeList()
+    {
+        $this->_setMockDB();
+        $expected = array(123 => 'scan 1', 234 => 'scan 2');
+        $this->_mockDB->setFakeTableData("mri_scan_type", $this->_mriInfo);
+        $this->_mockDB->setFakeTableData("files", $this->_fileInfo);
+        $this->assertEquals($expected, Utility::getScanTypeList());
+    }
+
+    /**
+     * Test that appendForwardSlash appends a forward slash to the given path.
+     * Also asserts that if the path already has a forward slash, it does nothing.
+     *
+     * @covers Utility::appendForwardSlash
+     * @return void
+     */
+    public function testAppendForwardSlash()
+    {
+        $path = "/php/libraries";
+        $pathWithSlash = "/php/libraries/";
+        $this->assertEquals(
+            $pathWithSlash,
+            Utility::appendForwardSlash($path)
+        );
+        $this->assertEquals(
+            $pathWithSlash,
+            Utility::appendForwardSlash($pathWithSlash)
+        );
+    }
+
+    /**
+     * Test that structureToPCRE returns the regex form of the given structure.
+     * This test covers the different cases of the function.
+     *
+     * @covers Utility::structureToPCRE
+     * @return void
+     */
+    public function testStructureToPCRE()
+    {
+        $structure = array(
+            'seq' => array(
+                0 => array('@' => array('type' => 'alpha',
+                                        'minLength' => '1',
+                                        'maxLength' => '5')
+                ),
+                1 => array('@' => array('type' => 'alphanumeric',
+                                        'length' => '2')
+                ),
+                2 => array('@' => array('type' => 'static'),
+                           '#' => '1-3'
+                ),
+                3 => array('@' => array('type' => 'set'),
+                           '#' => '1||3'
+                ),
+                4 => array('@' => array('type' => 'set'),
+                           '#' => '1-3'
+                ),
+            )
+        );
+        $this->assertEquals(
+            '/^[a-z]{1,5}[0-9a-z]{2,2}(1-3){1,1}(1||3){1,1}[1-3]{1,1}$/i',
+            Utility::structureToPCRE($structure)
+        );
+    }
+
+    /**
+     * Test structureToPCRE with the site and project abbreviations set
+     *
+     * @covers Utility::structureToPCRE
+     * @return void
+     */
+    public function testStructureToPCREWithAbbreviations()
```

Comment:
This was moved in #6766, I don't think this test is still valid here.

Reply:
@driusan Do you mean #6173? If so, I just saw that this was merged with master and I'll rebase and make the necessary changes. Thanks!